orthoinference/src/main/java/org/reactome/orthoinference/ReactionInferrer.java

```diff
@@ -52,87 +51,72 @@ public static void inferReaction(GKInstance reactionInst) throws Exception
 			infReactionInst.addAttributeValue(evidenceType, evidenceTypeInst);
 			infReactionInst.addAttributeValue(_displayName, reactionInst.getAttributeValue(_displayName));
 
-			// This function finds the total number of distinct proteins associated with an instance, as well as the number that can be inferred.
-			// Total proteins are stored in reactionProteinCounts[0], inferrable proteins in [1], and the maximum number of homologues for any entity involved in index [2].
-			// Reactions with no proteins/EWAS (Total = 0) are not inferred.
-			List<Integer> reactionProteinCounts = ProteinCountUtility.getDistinctProteinCounts(reactionInst);
-			int reactionTotalProteinCounts = reactionProteinCounts.get(0);
-			if (reactionTotalProteinCounts > 0) 
+			// Attempt to infer all PhysicalEntities associated with this reaction's Input, Output, CatalystActivity and RegulatedBy attributes.
+			// Failure to successfully infer any of these attributes will end inference for this reaction.
+			logger.info("Inferring inputs...");
+			if (inferReactionInputsOrOutputs(reactionInst, infReactionInst, input))
 			{
-				logger.info("Total protein count for RlE: " + reactionTotalProteinCounts);
-				String eligibleEventName = reactionInst.getAttributeValue(DB_ID).toString() + "\t" + reactionInst.getDisplayName() + "\n";	
-				// Having passed all tests/filters until now, the reaction is recorded in the 'eligible reactions' file, meaning inference is continued.
-				eligibleCount++;
-				Files.write(Paths.get(eligibleFilehandle), eligibleEventName.getBytes(), StandardOpenOption.APPEND);
-				// Attempt to infer all PhysicalEntities associated with this reaction's Input, Output, CatalystActivity and RegulatedBy attributes.
-				// Failure to successfully infer any of these attributes will end inference for this reaction.
-				logger.info("Inferring inputs...");
-				if (inferReactionInputsOrOutputs(reactionInst, infReactionInst, input))
+				logger.info("Inferring outputs...");
+				if (inferReactionInputsOrOutputs(reactionInst, infReactionInst, output))
```

Comment:
It may be clearer to use guard clauses rather than nest. 

https://stackoverflow.com/a/4887280/2295778
https://refactoring.guru/replace-nested-conditional-with-guard-clauses

```
// Attempt to infer all PhysicalEntities associated with this reaction's Input, Output, CatalystActivity and RegulatedBy attributes.
// Failure to successfully infer any of these attributes will end inference for this reaction.
logger.info("Inferring inputs...");
if (!(inferReactionInputsOrOutputs(reactionInst, infReactionInst, input)))
{
	logger.info("Input inference unsuccessful -- terminating inference for " + reactionInst);
	return;
}

logger.info("Inferring outputs...");
if (!(inferReactionInputsOrOutputs(reactionInst, infReactionInst, output)))
{
	logger.info("Output inference unsuccessful -- terminating inference for " + reactionInst);
	return;
}

logger.info("Inferring catalysts...");
if (!(inferReactionCatalysts(reactionInst, infReactionInst)))
{
	logger.info("Catalyst inference unsuccessful -- terminating inference for " + reactionInst);
	return;
}
```